Weather Alert/CoreData/DataController.swift

```diff
@@ -80,6 +80,33 @@ class DataController {
         }
         return true
     }
+}
+
+extension DataController {
     
+    var dataController: DataController { return AppShared.instances.dataController }
+    
+    func store() {
+        do {
+            try dataController.managedObjectContext.save()
+        } catch {
+            fatalError("Failure to save context: \(error)")
+        }
+    }
     
+    func discard() {
+        dataController.managedObjectContext.rollback()
+    }
+    
+    func requestCities() -> [City] {
```

Comment:
Class specific implementation should be elsewhere. This is the core of the class and should be stripped from unrelated/specific classes.